### PR TITLE
LPS-123918 Traffic SOCIAL detail backend mock

### PR DESCRIPTION
**Pull request by darquesdev**
```javascript
[
   {
      "name":"referral",
      "share":"0,0",
      "title":"Referral",
      "value":0,
      "helpMessage":"This is the number of page views generated by people coming to your page from other sites which are not search engine pages or social sites."
   },
   {
      "name":"social",
      "share":"90,0",
      "details":{
         "referringPages":[
            {
               "url":"https://www.linkedin.com/company/liferay-inc-/?originalSubdomain=es",
               "traffic":525
            },
            {
               "url":"https://twitter.com/liferay_es?lang=es",
               "traffic":455
            },
            {
               "url":"https://www.facebook.com/liferay/",
               "traffic":100
            },
            {
               "url":"https://www.youtube.com/channel/UC3Ya5ZhK1SZocVdA_5QwGXg",
               "traffic":100
            },
            {
               "url":"https://medium.com/developer-relations/liferay-hackday-2017-7c503d55276b",
               "traffic":54
            }
         ],
         "referringSocialMedia":[
            {
               "referringPages":[
                  {
                     "url":"https://www.linkedin.com/company/liferay-inc-/?originalSubdomain=es",
                     "traffic":525
                  },
                  {
                     "url":"https://www.linkedin.com/company/liferay-cloud/?originalSubdomain=es",
                     "traffic":112
                  }
               ],
               "name":"linkedin",
               "title":"LinkedIn",
               "traffic":1256
            },
            {
               "referringPages":[
                  {
                     "url":"https://twitter.com/liferay_es?lang=es",
                     "traffic":525
                  },
                  {
                     "url":"https://twitter.com/liferay?lang=es",
                     "traffic":444
                  },
                  {
                     "url":"https://twitter.com/dgomezg?lang=es",
                     "traffic":412
                  },
                  {
                     "url":"https://twitter.com/LiferayDEVCON",
                     "traffic":400
                  },
                  {
                     "url":"https://twitter.com/Saray_dg",
                     "traffic":396
                  },
                  {
                     "url":"https://twitter.com/liferaydesign",
                     "traffic":350
                  },
                  {
                     "url":"https://twitter.com/LiferayTango",
                     "traffic":200
                  },
                  {
                     "url":"https://twitter.com/liferayeng",
                     "traffic":145
                  },
                  {
                     "url":"https://twitter.com/LiferayLearning",
                     "traffic":112
                  },
                  {
                     "url":"https://twitter.com/liferayndragons",
                     "traffic":55
                  }
               ],
               "name":"twitter",
               "title":"Twitter",
               "traffic":1165
            },
            {
               "referringPages":[
                  {
                     "url":"https://www.facebook.com/liferay/",
                     "traffic":125
                  },
                  {
                     "url":"https://www.facebook.com/liferay.tango.9",
                     "traffic":5
                  }
               ],
               "name":"facebook",
               "title":"Facebook",
               "traffic":129
            },
            {
               "referringPages":[
                  {
                     "url":"https://www.youtube.com/channel/UC3Ya5ZhK1SZocVdA_5QwGXg",
                     "traffic":99
                  },
                  {
                     "url":"https://www.youtube.com/watch?v=PDreyYdq-Ic&ab_channel=Liferay",
                     "traffic":78
                  },
                  {
                     "url":"https://www.youtube.com/watch?v=dY5MXiYVWvg&ab_channel=AlbertCoronado",
                     "traffic":55
                  }
               ],
               "name":"youtube",
               "title":"YouTube",
               "traffic":150
            },
            {
               "referringPages":[
                  {
                     "url":"https://medium.com/@KNOWARTH/redefining-digital-experience-how-liferay-portal-development-does-it-bd3bae704781",
                     "traffic":12
                  },
                  {
                     "url":"https://www.whatsapp.com/features/",
                     "traffic":5
                  },
                  {
                     "url":"https://www.imqq.com/",
                     "traffic":4
                  },
                  {
                     "url":"https://www.tumblr.com/",
                     "traffic":3
                  }
               ],
               "name":"others",
               "title":"Others",
               "traffic":30
            }
         ]
      },
      "title":"Social",
      "value":5532,
      "helpMessage":"This is the number of page views generated by people coming to your page from social sites."
   },
   {
      "name":"paid",
      "share":"0,0",
      "title":"Paid",
      "value":0,
      "helpMessage":"This is the number of page views generated by people that find your page through Google AdWords."
   },
   {
      "name":"direct",
      "share":"0,0",
      "title":"Direct",
      "value":0,
      "helpMessage":"This is the number of page views generated by people arriving directly to your page."
   },
   {
      "name":"organic",
      "share":"0,0",
      "title":"Organic",
      "value":0,
      "helpMessage":"This is the number of page views generated by people coming from a search engine."
   }
]
```

### CODE REVIEW

**Comment by darquesdev** [comment]
Hi @dgarciasarai! Here's we have our backend mock. Let me know if you need any change ;)

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/portlet/action/GetTrafficSourcesMVCResourceCommand.java [hunk 1]
```diff
@@ -135,6 +140,96 @@ protected void doServeResource(
 		}
 	}
 
+	private SocialMediaTrafficDetails _getSocialMediaTrafficDetails() {
+		return new SocialMediaTrafficDetails(
+			Arrays.asList(
```
Comment (cgoncas) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/portlet/action/GetTrafficSourcesMVCResourceCommand.java:145]: if you want to add this mock, that is totally fine, please, use a feature flag to activate it :)

**Comment by dgarciasarai** [comment]
Hey @darquesdev, for `referringPages` we should retrieve 10 URLs, and on the frontend side shows 5 URLs with "View More" button. If we click on it, the button changes to "View Less" and shows 10 URLs. Thanks!

**Comment by darquesdev** [comment]
Hi @dgarciasarai, I've updated the mock with the new requirements:
- Remove the referral sections
- Don't use `detail` property
- Use `trafficAmount` instead of `traffic`

**Comment by darquesdev** [comment]
```json
[
    {
       "name":"referral",
       "share":"0,0",
       "title":"Referral",
       "value":0,
       "helpMessage":"This is the number of page views generated by people coming to your page from other sites which are not search engine pages or social sites."
    },
    {
       "name":"social",
       "share":"90,0",
       "title":"Social",
       "referringSocialMedia":[
          {
             "name":"linkedin",
             "trafficAmount":1256,
             "title":"LinkedIn"
          },
          {
             "name":"twitter",
             "trafficAmount":1165,
             "title":"Twitter"
          },
          {
             "name":"facebook",
             "trafficAmount":229,
             "title":"Facebook"
          },
          {
             "name":"youtube",
             "trafficAmount":150,
             "title":"YouTube"
          },
          {
             "name":"other",
             "trafficAmount":30,
             "title":"Other"
          }
       ],
       "value":5532,
       "helpMessage":"This is the number of page views generated by people coming to your page from social sites."
    },
    {
       "name":"paid",
       "share":"0,0",
       "title":"Paid",
       "value":0,
       "helpMessage":"This is the number of page views generated by people that find your page through Google AdWords."
    },
    {
       "name":"direct",
       "share":"0,0",
       "title":"Direct",
       "value":0,
       "helpMessage":"This is the number of page views generated by people arriving directly to your page."
    },
    {
       "name":"organic",
       "share":"0,0",
       "title":"Organic",
       "value":0,
       "helpMessage":"This is the number of page views generated by people coming from a search engine."
    }
 ]
```

**Comment by cgoncas** [comment]
Hey @dgarciasarai and @darquesdev !!!

Could you merge this mock with the final backend/frontend solution, so QA can test it?

Thanks!!!

**Comment by darquesdev** [comment]
I'll prepare a single PR with both REFERRAL and SOCIAL traffic mocks

**Comment by darquesdev** [comment]
https://github.com/liferay-tango/liferay-portal/pull/641

**Pull request closed without merging**